fvtest/tril/test/ParserTest.cpp

```diff
@@ -26,26 +26,33 @@
 #define ASSERT_NULL(pointer) ASSERT_EQ(nullptr, (pointer))
 #define ASSERT_NOTNULL(pointer) ASSERT_TRUE(nullptr != (pointer))
 
-TEST(ParserTest, SingleNodeWithJustName) {
-    auto trees = parseString("(nodeName)");
 
-    ASSERT_NOTNULL(trees);
-    ASSERT_STREQ("nodeName", trees->getName());
-    ASSERT_NULL(trees->getArgs());
-    ASSERT_NULL(trees->getChildren());
-    ASSERT_NULL(trees->next);
-}
+typedef std::pair<const char *, const char *> TestParamWithExpectedResultString;
 
-TEST(ParserTest, SingleCommandNodeWithJustName) {
-    auto trees = parseString("(@commandName)");
+class SingleNodeWithName : public testing::TestWithParam<TestParamWithExpectedResultString> {};
+
+TEST_P(SingleNodeWithName, VerifyNode) {
+    auto trees = parseString(GetParam().first);
 
     ASSERT_NOTNULL(trees);
-    ASSERT_STREQ("@commandName", trees->getName());
+    ASSERT_STREQ(GetParam().second, trees->getName());
     ASSERT_NULL(trees->getArgs());
     ASSERT_NULL(trees->getChildren());
     ASSERT_NULL(trees->next);
 }
 
+// The generated tests from this instantiation will have the following names:
+// ParserTest/SingleNodeWithName.VerifyNode/0 for TestParamWithExpectedResultString("(nodeName)", "nodename") ..
+// With the convention being:
+// InstantiationName/FooTest.DoesBlah/n for val_n
```

Comment:
Not sure this comment is needed, but I guess it doesn't hurt to have around.

Reply:
I'm okay either way, don't think it matters that much. @mgaudet, thoughts?

Reply:
I'd say pull the comment -- just because it's documenting the expected behaviour of GoogleTest, and so isn't adding a huge amount value for experienced readers (though, I understand how it would help people new to the project; we just can't optimize the code for them, or else the code would be so thick with explanatory comment that it would be difficult to see what's going on)